include/irr/asset/IMeshPacker.h

```diff
@@ -0,0 +1,123 @@
+#ifndef __IRR_I_MESH_PACKER_H_INCLUDED__
+#define __IRR_I_MESH_PACKER_H_INCLUDED__
+
+namespace irr
+{
+namespace asset
+{
+
+namespace meshPackerUtil
+{
+    struct AllocationParams
+    {
+        size_t indexBuffSupportedSizeInBytes                 = 2147483648ull;      /*  2GB*/
+        size_t vertexBuffSupportedSizeInBytes                = 2147483648ull;      /*  2GB*/
+        size_t perInstanceVertexBuffSupportedSizeInBytes     = 33554432ull;        /* 32MB*/
+        size_t MDIDataBuffSupportedSizeInBytes               = 2147483648ull;      /*  2GB*/
+        size_t vertexBufferMinAllocSize                      = 512ull;             /* 512B*/
+        size_t indexBufferMinAllocSize                       = 256ull;             /* 256B*/
+        size_t perInstanceVertexBufferMinAllocSize           = 32ull;              /*  32B*/
+        size_t MDIDataBuffMinAllocSize                       = 32ull;              /*  32B*/
+    };
+
+    template <typename MeshBufferType>
+    struct PackedMeshBuffer
+    {
+        core::smart_refctd_ptr<MeshBufferType> packedMeshBuffer = nullptr;
+    };
+
+    struct PackedMeshBufferData
+    {
+        uint32_t mdiParameterByteOffset; // add to `CCPUMeshPacker::getMultiDrawIndirectBuffer()->getPointer() to get `DrawElementsIndirectCommand_t` address
```

Comment:
call it `offset` not `ByteOffset` because you will be doing
```cpp
reinterpret_cast<MDIStructType*>(CCPUMeshPacker::getMultiDrawIndirectBuffer()->getPointer())+PackedMeshBufferData::mdiParameterOffset`
```
NOT

```cpp
reinterpret_cast<MDIStructType*>(reinterpret_cast<uint8_t*>(CCPUMeshPacker::getMultiDrawIndirectBuffer()->getPointer())+PackedMeshBufferData::mdiParameterByteOffset)`
```